inst/rmarkdown/templates/vaccination/skeleton/skeleton.Rmd

```diff
@@ -3,11 +3,184 @@ title: "Vaccination survey"
+vaccine_data$l1_q10_age_yr <- as.numeric(vaccine_data$l1_q10_age_yr)
+vaccine_data$l1_q55_age_mth <- as.numeric(vaccine_data$l1_q55_age_mth)
+
+
+# Create an age group in months from the year variable
+vaccine_data$age_group <- vaccine_data$l1_q10_age_yr
+
+vaccine_data$age_group[vaccine_data$l1_q55_age_mth < 12 | vaccine_data$l1_q10_age_yr == 0] <- "0-11"
+vaccine_data$age_group[vaccine_data$age_group >= 1 & vaccine_data$age_group < 2] <- "12-23"
+vaccine_data$age_group[vaccine_data$age_group >= 2 & vaccine_data$age_group < 3] <- "24-35"
+vaccine_data$age_group[vaccine_data$age_group >= 3 & vaccine_data$age_group < 4] <- "36-47"
+vaccine_data$age_group[vaccine_data$age_group >= 4 & vaccine_data$age_group < 5] <- "48-59"
+vaccine_data$age_group[vaccine_data$age_group >= 5 & vaccine_data$age_group < 15] <- "60+"
+
+
+# Additional variable name cleaning and creation of vaccination status binary variable
+vaccine_data_clean <- vaccine_data %>%
+                      mutate(sex = l1_q5_sex,
+                             vaccination_status = l1_q2_vaccine_9months,
+                             area = l0_q4_settlement,
+                             cluster = l0_q77_what_is_the_cluster_number,
+                             vaccination_status_simple = ifelse(vaccination_status == "Yes - Verbal"
+                                                                | vaccination_status == "Yes - Vaccination card", "Yes",
+                                                                ifelse(vaccination_status == "No" |
+                                                                         vaccination_status == "Don't know", "No", NA)))
+
+
+```
+
+
+``` {r survey design}
+# TODO - figure out weights to use and design effect
+
+# we first need to create a survey design that reflects the study
+survey_design <- svydesign(
+  ids = ~ cluster, # cluster id is stored as cluster
+  weights = weights, #sampling weights applied to this example
+  data = vaccine_data_clean
+)
+```
+
+
+
+## Characteristics of the sampled children
+
+Children included in the survey by sex
+```{r participant description by sex}
+
+# get counts and proportions of participants by sex
+descriptive(vaccine_data_clean, "sex") %>%
+  # change table column names 
+  # rename( new variable name = old variable name)
+  rename("Sex" = sex, "Number of participants (n)" = n,"Proportion (%)" =  prop) %>% 
+  kable(digits = 2)
+```
+
+
+Children included in the survey by age group
+```{r participant description by age group}
+
+# get counts and proportions of participants by age group
+descriptive(vaccine_data_clean, "age_group") %>%
+  # change table column names 
+  # rename( new variable name = old variable name)
+  rename("Age group" = age_group, "Number of participants (n)" = n,"Proportion (%)" =  prop) %>% 
+  kable(digits = 2)
+```
+
+Age pyramid
+
+```{r age_pyramid, warning=FALSE}
+# plot age pyramid 
+plot_age_pyramid(vaccine_data_clean, age_group = "age_group", split_by = "sex") + 
```

Comment:
Zhian has updated to be able to create age pyramid as proportion of total population.
However we will need to create another function to allow weighted proportions!